### PR TITLE
Management interface

### PR DESCRIPTION
**Pull request by chriseth**
This PR adds several RPC endpoints provided by geth but missing from eth.
Especially, it makes `eth` work as a backend for `geth attach`.

EXPOSES INTERNAL DATA

This will expose some functions via rpc that were only accessible using the session key.
This is not a problem via the unix socket (the session key is readable from a file anyway), but it might be via http, so exposing the admin interface via http has to be explicitly requested.

Fixes: https://github.com/ethereum/webthree-umbrella/issues/161

### CODE REVIEW

File: libweb3jsonrpc/AdminEth.cpp [hunk 1]
```diff
@@ -250,3 +237,33 @@ Json::Value AdminEth::admin_eth_getReceiptByHashAndIndex(std::string const& _blo
...
+	return true;
+}
+
+bool AdminEth::miner_stop()
+{
+	m_eth.stopSealing();
+	return true;
+}
+
+bool AdminEth::miner_setEtherbase(std::string const& _uuidOrAddress)
```
Comment (LianaHus) [inline libweb3jsonrpc/AdminEth.cpp]: no need for std::

File: libweb3jsonrpc/Whisper.h [hunk 2]
```diff
@@ -43,7 +43,11 @@ class Whisper: public WhisperFace
 public:
 	// TODO: init with whisper interface instead of webthreedirect
 	Whisper(WebThreeDirect& _web3, std::vector<dev::KeyPair> const& _accounts);
-	
+	virtual std::pair<std::string, std::string> implementedModule() const override
```
Comment (LianaHus) [inline libweb3jsonrpc/Whisper.h]: IMHO you could typedef std::pair<std::string, std::string>

File: libweb3jsonrpc/AdminEth.cpp (hunk 1 above)
Comment (LianaHus) [inline libweb3jsonrpc/AdminEth.cpp]: I just noticed that this file contains a lot of std::'s

**Comment by winsvega** [comment]
geth attach wont work

```
Fatal: Unable to initialize console - METHOD_NOT_FOUND: The method being requested is not available on this server
```

**Comment by LianaHus** [comment]
looks good for me. @winsvega did you test it?

**Comment by winsvega** [comment]
still 
Fatal: Unable to attach to geth node - dial unix /home/wins/Ethereum/testnet/ethnode1/geth.ipc: no such file or directory

geth fetched from branch master
umbrella from latest develop. webthree from chriseth/managmentInterface
running eth with --json-rpc

File: libweb3jsonrpc/AdminNet.cpp [hunk 3]
```diff
@@ -54,4 +50,36 @@ Json::Value AdminNet::admin_net_nodeInfo(std::string const& _session)
...
+	ret["ports"] = Json::objectValue;
+	// Both ports are equal as of 2016-02-04, migt change later
+	ret["ports"]["discovery"] = i.port;
+	ret["ports"]["listener"] = i.port;
+	ret["ip"] = i.address;
+	ret["listenAddr"] = i.address + ":" + toString(i.port);
+	ret["id"] = i.id.hex();
+	ret["enode"] = i.enode();
+	ret["protocols"] = Json::objectValue;
+	ret["protocols"]["eth"] = Json::objectValue; //@todo fill with information
```
Comment (LianaHus) [inline libweb3jsonrpc/AdminNet.cpp:69]: here :)

File: libweb3jsonrpc/JsonHelper.cpp [hunk 4]
```diff
@@ -62,16 +67,17 @@ namespace p2p
 
 Json::Value toJson(p2p::PeerSessionInfo const& _p)
 {
+	//@todo localAddress
+	//@todo protocols
```
Comment (LianaHus) [inline libweb3jsonrpc/JsonHelper.cpp:71]: here

File: libweb3jsonrpc/AdminNet.cpp (hunk 3 above)
Comment (chriseth) [inline libweb3jsonrpc/AdminNet.cpp:69]: erm, ok, yeah, I won't fix that for now :-)

File: libweb3jsonrpc/JsonHelper.cpp (hunk 4 above)
Comment (chriseth) [inline libweb3jsonrpc/JsonHelper.cpp:71]: Won't fix. Some of this information needs a refactoring in other codebases.

**Comment by chriseth** [comment]
as @winsvega discovered, we have to move the ipc connector if the datadir is changed.

**Comment by chriseth** [comment]
Moving the socket with the data dir is addressed in https://github.com/ethereum/webthree/pull/127

**Comment by chriseth** [comment]
As this PR has already been marked looksgood and only got issues by me, I will  merge.

**Comment by winsvega** [comment]
geth attach now works. thanks!

**Pull request merged**